### DISCUSSION TITLE
RTL is not consistent cross application

### DISCUSSION THREAD

**Discussion by avma** [pawelsalawa/sqlitestudio#3465] [General]
Hi,

I completed translation both coreSQLiteStudio and guiSQLiteStudio to Hebrew, while testing the translation, I noticed that Right To Left is not consistent across all windows. 
In most cases it work just fine. problem starts when a phrase combine both TRL and LTR characters. for Instance: DDL (and such) should not translate to Hebrew. Now when it appears at the end or beginning of the phrase, (for example) "DDL History size:" in hebrew it will show: "History size DDL" (in hebrew characters...) and vise versa. For this, there is an easy work around by just putting "DDL" to the end or beginning of the hebrew phrase to make it show right (not convenient but doable), but when in appears in mid phrase such: "Don't show DDL preview dialog..." in hebrew it will show: "DDL preview dialog Don't show ..." which does not have any sense and the workaround manipulation will not work.
Furthermore, it gets even worse when where HTML Attribute are used 

> `<p>, <p/>....`

Do you have any suggestion/idea of how to  tackle this?
My regards
Avi

**Comment by pawelsalawa** [comment]
I'm sorry, but I'm not really familiar with RTL languages. From what I understand - when you translate a message, you should do it in a way that is correct in Hebrew, considering that words order will be reversed (right?). I mean you translate a whole Hebrew message, so you get to decide order of words, don't you?

**Comment by avma** [comment]
True, but when one or (worse) more LTR char gets in the middle of the sentence, it will mess up the order of the words hence all the sentence gets fuzzy.
This, I believe will be the case with all RTL languages.

**Comment by pawelsalawa** [comment]
Could you please given an example of original, English message, then how it is translated by you (you can put Hebrew characters here) and then screenshot of how it looks in the app? I'm trying to understand the problem.

**Comment by avma** [comment]
Attached are screen shots of two examples: you can see original text, the way it should look in hebrew (in po edit) and the way it shows  on sqlitestudio. To make it easier (the numbers for hebrew words are in the sequence they should be). 


![screenshot from 2018-12-04 16-59-51](https://user-images.githubusercontent.com/7680483/49453077-14044700-f7eb-11e8-8da6-5698ffae49b5.png)
![screenshot from 2018-12-04 16-58-23](https://user-images.githubusercontent.com/7680483/49453078-14044700-f7eb-11e8-9df8-a71964d92419.png)
![screenshot from 2018-12-04 16-57-09](https://user-images.githubusercontent.com/7680483/49453079-14044700-f7eb-11e8-847e-7fe6cf13a354.png)

**Comment by avma** [comment]
one more example with html involved: **The first on is not OK!**
![screenshot from 2018-12-04 17-53-55](https://user-images.githubusercontent.com/7680483/49454387-b6bdc500-f7ed-11e8-90be-6c9db27f9f87.png)
**This is how it should look:**
![screenshot from 2018-12-04 17-53-12](https://user-images.githubusercontent.com/7680483/49454388-b7565b80-f7ed-11e8-9605-fa3312c5ecf4.png)

**Comment by pawelsalawa** [comment]
Looks like it may be related to https://bugreports.qt.io/browse/QTBUG-61529?attachmentViewMode=list

Do you think so too?

**Comment by avma** [comment]
Yep, same problem...
Well, until this is fixed (if ever) i found a workaround to manipulate the broken hebrew sentences with this ancient tool [http://www.pixiesoft.com/flip/](url) not very convenient but works for most cases...

**Comment by pawelsalawa** [comment]
Thanks for sharing this. I will monitor original Qt bug.
